include/zephyr/sys/ktime.h

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) 2023 Zephyr Project
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Representation and conversion of nanosecond resolution time and
+ * timestamps.
+ *
+ * Declare types and utilities to represent high resolution time and timers used
+ * by the kernel as well as by applications.
+ *
+ * Inspired by
+ * https://github.com/torvalds/linux/blob/master/include/linux/ktime.h and
+ * https://github.com/torvalds/linux/blob/master/[tools/]include/linux/time64.h
+ */
+
+#ifndef ZEPHYR_INCLUDE_SYS_KTIME_H_
+#define ZEPHYR_INCLUDE_SYS_KTIME_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/* number of nsec per usec */
+#define NSEC_PER_USEC 1000U
+
+/* number of nsec per msec */
+#define NSEC_PER_MSEC 1000000U
+
+/* number of microseconds per millisecond */
+#define USEC_PER_MSEC 1000U
+
+/* number of milliseconds per second */
+#define MSEC_PER_SEC 1000U
+
+/* number of microseconds per second */
+#define USEC_PER_SEC ((USEC_PER_MSEC) * (MSEC_PER_SEC))
+
+/* number of nanoseconds per second */
+#define NSEC_PER_SEC ((NSEC_PER_USEC) * (USEC_PER_MSEC) * (MSEC_PER_SEC))
```

Comment:
Use javadoc style comments /** ... */

Reply:
@kartben I forgot the @cond INTERNAL_HIDDEN wrapper around these constants (now added). I also made them /** ... */ comments now. Not sure whether this is still required for hidden constants, though.

@andyross The reason is that you were not much in favor of exposing those constants too much - at least not as a well-published API that incentivates further proliferation of their use.

Reply:
Ya if they're hidden you're fine dropping the Javadoc style comments. That would actually help make it even clearer they're not "API" (which I didn't know if they were when originally commenting)

Reply:
Ya if they're hidden you're fine dropping the Javadoc style comments. That would actually help make it even clearer they're not "API" (which I didn't know if they were when originally commenting)

Reply:
Makes sense. Returned to inline /* ... */ comments. :-)